daft/expressions/expressions.py

```diff
@@ -1930,6 +1930,19 @@ def url_parse(self) -> Expression:
         f = native.get_function_from_registry("url_parse")
         return Expression._from_pyexpr(f(self._expr))
 
+    def explode(self) -> Expression:
+        """Explode a list expression.
+
+        A row is created for each item in the lists, and other columns are broadcasted to match.
+
+        If exploding multiple columns at once, all list lengths must match.
```

Comment:
We should definitely add some doctest examples here. 

Ideally we should have some where we are 
1. exploding a column with a broadcast of others `select(a, b.explode())`
2. exploding multiple columns `select(a.explode(), b.explode())`
3. exploding multiple columns that wont work `select(a.explode(), b_with_different_len.explode())`